runtime/wasm/src/asc_abi/class.rs

```diff
@@ -241,8 +238,7 @@ impl AscValue for EnumPayload {}
 
 impl From<EnumPayload> for i32 {
     fn from(payload: EnumPayload) -> i32 {
-        // This is just `i32::from_bytes` which is unstable.
-        unsafe { ::std::mem::transmute::<u32, i32>(payload.0 as u32) }
+        payload.0 as i32
```

Comment:
These ones had a reason to be the way they were. A cast has overflow semantics which under some build configurations, such as debug, may panic. Here we just want to transmute the bytes with no arithmetic consequence, so transmute is more correct than a cast. `from_bytes` is a safe way of doing that transmute that we can use now.